engines/stark/ui/world/gamewindow.cpp

```diff
@@ -238,4 +241,25 @@ void GameWindow::reset() {
 	_objectRelativePosition.y = 0;
 }
 
+void GameWindow::onScreenChanged() {
+	// May be called when resources have not been loaded
+	if (!StarkGlobal->getCurrent()) {
+		return;
+	}
+
+	Common::Array<Resources::Layer *> layers = StarkGlobal->getCurrent()->getLocation()->listLayers();
+
+	VisualText *text = nullptr;
+	Gfx::RenderEntryArray renderEntries;
+	for (uint i = 0; i < layers.size(); ++i) {
+		renderEntries = layers[i]->listRenderEntries();
+		for (uint j = 0; j < renderEntries.size(); ++j) {
+			text = renderEntries[j]->getText();
```

Comment:
Texts in not only the current layer but all the layers within the same location need to reset their textures when the resolution has changed, otherwise, the user may change the screen, go back to a previous layer and find size-unmatched texts.